src/formnetworks/abstractformnetwork.jl

```diff
@@ -20,6 +20,13 @@ function SimilarType.similar_type(f::AbstractFormNetwork)
   return typeof(tensornetwork(f))
 end
 
+# TODO: Use `NamedGraphs.GraphsExtensions.parent_graph_type`.
+function data_graph_type(G::Type{<:AbstractFormNetwork})
+  return data_graph_type(fieldtype(G, :tensornetwork))
+end
```

Comment:
I realize this was just moved in this PR but I noticed there is something off with this definition.

In general, it is best to not define definitions on abstract types that assume subtypes have a certain field (generic functions defined on abstract types should only depend on how objects _behave_, not how their data is stored).

Does this need to be in the type domain? I.e. an alternative could be:
```julia
function data_graph_type(f::AbstractFormNetwork)
  return data_graph_type(tensornetwork(g))
end
```
Other alternatives would be, to make the tensor network type a type parameter of `AbstractFormNetwork`, or expect subtypes to define this function themselves.

Reply:
Good point, yeah this was assuming a field that may or may not exist.

I think moving it to the type domain should work, I just pushed that change.